docs/tutorials/07_grover_examples.ipynb

```diff
@@ -0,0 +1,250 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Grover's algorithm examples\n",
+    "\n",
+    "This notebook has examples demonstrating how to use the Qiskit Algorithms [Grover](https://qiskit.org/ecosystem/algorithms/stubs/qiskit_algorithms.Grover.html) search algorithm, with different oracles."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Finding solutions to 3-SAT problems\n",
+    "\n",
+    "Let's look at an example 3-Satisfiability (3-SAT) problem and walk-through how we can use Quantum Search to find its satisfying solutions. 3-SAT problems are usually expressed in [Conjunctive Normal Forms (CNF)](https://en.wikipedia.org/wiki/Conjunctive_normal_form) and written in the [DIMACS-CNF](http://www.satcompetition.org/2009/format-benchmarks2009.html) format. For example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_3sat_instance = '''\n",
+    "c example DIMACS-CNF 3-SAT\n",
+    "p cnf 3 5\n",
+    "-1 -2 -3 0\n",
+    "1 -2 3 0\n",
+    "1 2 -3 0\n",
+    "1 -2 -3 0\n",
+    "-1 2 3 0\n",
+    "'''"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The CNF of this 3-SAT instance contains 3 variables and 5 clauses:\n",
+    "\n",
+    "$(\\neg v_1 \\vee \\neg v_2 \\vee \\neg v_3) \\wedge (v_1 \\vee \\neg v_2 \\vee v_3) \\wedge (v_1 \\vee v_2 \\vee \\neg v_3) \\wedge (v_1 \\vee \\neg v_2 \\vee \\neg v_3) \\wedge (\\neg v_1 \\vee v_2 \\vee v_3)$\n",
+    "\n",
+    "It can be verified that this 3-SAT problem instance has three satisfying solutions:\n",
+    "\n",
+    "$(v_1, v_2, v_3) = (T, F, T)$ or $(F, F, F)$ or $(T, T, F)$\n",
+    "\n",
+    "Or, expressed using the DIMACS notation:\n",
+    "\n",
+    "`1 -2 3`, or `-1 -2 -3`, or `1 2 -3`.\n",
+    "\n",
+    "With this example problem input, we then create the corresponding `oracle` for our `Grover` search. In particular, we use the `PhaseOracle` component, which supports parsing DIMACS-CNF format strings and constructing the corresponding oracle circuit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\"The 'tweedledum' library is required to use 'BooleanExpression'. You can install it with 'pip install tweedledum'.\"\n"
```

Comment:
> The 'tweedledum' library is required to use 'BooleanExpression'. You can install it with 'pip install tweedledum'

We should probably add `tweedledum` to the requirements-dev so that this tutorial runs properly without raising an issue.

Reply:
Agreed. This is how we do it in Terra:

```
# Tweedledum is unmaintained and its existing Mac wheels are unreliable. If you
# manage to get a working install on a Mac the functionality should still work,
# but as a convenience this file won't attempt the install itself.
tweedledum; python_version<'3.11' and platform_system!="Darwin"
```

Reply:
Here we only need it for when the tutorials are run under ubuntu; and since we publish from a 3.8 run the fact that there is no wheel elsewhere is not a problem in that regard. Maybe annoying otherwise of course!